HouseRules_Essentials/Rules/LevelSequenceOverriddenRule.cs

```diff
@@ -27,37 +19,53 @@ public sealed class LevelSequenceOverriddenRule : Rule, IConfigWritable<List<str
         public LevelSequenceOverriddenRule(List<string> adjustments)
         {
             _adjustments = adjustments;
-            _originals = new List<string>();
         }
 
         public List<string> GetConfigObject() => _adjustments;
 
-        protected override void OnPostGameCreated(GameContext gameContext)
+        protected override void OnPreGameCreated(GameContext gameContext)
+        {
+            ReplaceExistingProperties(_adjustments, gameContext);
+        }
+
+        private static void Patch(Harmony harmony)
         {
-            _originals = ReplaceExistingProperties(_adjustments);
+            harmony.Patch(
+                original: AccessTools.Method(
+                    typeof(LevelSequenceConfiguration), "GetSequenceDefinition"),
+                prefix: new HarmonyMethod(typeof(LevelSequenceOverriddenRule), nameof(LevelSequenceConfiguration_GetSequenceDefinition_Prefix)));
         }
 
-        protected override void OnDeactivate(GameContext gameContext)
+        /// <remarks>
+        /// Sets a safe sequence definition even if the active game type does not have one that extends to the current level.
+        /// </remarks>
+        private static bool LevelSequenceConfiguration_GetSequenceDefinition_Prefix(ref SequenceDefinition __result, int index, LevelSequence.GameType gameType)
```

Comment:
`BoardGameActionGoToNextLevel.StartNextLevel()` was calling `levelManager.GetCurrentLevelObject()` which was calling `levelSequenceConfiguration.GetSequenceDefinition()` which would throw an exception if the current level index was out of bounds of what the predefined gametype had defined.

This method patches `levelSequenceConfiguration.GetSequenceDefinition()` so that if the level index is out of bounds, we return some sequence definition within bounds (the last sequencedefinition if the last level, or the 3rd-to-last definition otherwise).

---

At first glance, the only real purpose of `levelSequenceConfiguration.GetSequenceDefinition()` in `BoardGameActionGoToNextLevel.StartNextLevel()` is to know whether or not the level definition is a shop stage.  However, shop levels seem to load fine even if the returned levelSequence is almost arbitrary (as I've done here).

In case we experience issues with loading shops as part of this rule, chances are we need to re-patch this method more appropriately.  But seems fine thus far.